### PR TITLE
OCPBUILD-9: Adds capabilities for builder & deployer rolebindings controller

### PR DESCRIPTION
**Pull request by apoorvajagtap**
Related to: https://github.com/openshift/api/pull/1770
Required for: https://github.com/openshift/openshift-controller-manager/pull/289
https://github.com/openshift/openshift-controller-manager/pull/296

### CODE REVIEW

**Comment by adambkaplan** [comment]
@apoorvajagtap while we are waiting on OCPBUGS-33041 to be fixed, can you please re-do this pull request as two commits?

- Commit 1: title it `bump(openshift/api): Names for New OCM Controllers`, only include the go.mod, go.sum, and vendor/* updates.
- Commit 2: updates to controller code. Adhere to these [seven rules for great commit messages](https://cbea.ms/git-commit/#seven-rules)

**Comment by adambkaplan** [comment]
/hold

Needs https://github.com/openshift/openshift-controller-manager/pull/296 to merge first.

**Comment by adambkaplan** [comment]
/retest

Build tests should work now.

**Review by adambkaplan — APPROVED** [review]
/approve

Needs additional test case updates in order to :+1: for merge (unit tests currently failing).

**Comment by adambkaplan** [comment]
/hold cancel

Removing the hold since the controller manager changes have merged. This will still need to wait on [OCPBUGS-33378](https://issues.redhat.com/browse/OCPBUGS-33378) to be addressed before tide can merge this.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (apoorvajagtap) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:1]: @adambkaplan As all these three capabilities are enabled by default, and I believe [ControllersEnabled testCase](https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go#L122) covers such scenario, so, do we still need separate test cases for each Cap enabled?
IIUC, adding “ImageRegistry Cap” to current [ControllersEnabled testCase](https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go#L122) with additional testCases for disabled capabilities would suffice. 
Please let me know in case I am missing anything.

Comment (apoorvajagtap) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:1]: @adambkaplan As all these three capabilities are enabled by default, and I believe [ControllersEnabled testCase](https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go#L122) covers such scenario, so, do we still need separate testCases for each Cap enabled?
IIUC, adding “ImageRegistry Cap” to current [ControllersEnabled testCase](https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go#L122) with additional testCases for disabled capabilities would suffice.
Please let me know in case I am missing anything.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (adambkaplan) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:158]: /cc @ingvagabund  

Verify that these controllers should be disabled if DeploymentConfig is turned off.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (adambkaplan) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:170]: /cc @flavianmissi 

Verify that the image puller roles should be disabled if ImageRegistry capability is diabled

**Review by adambkaplan — COMMENTED** [review]
LGTM from me.

Adding leads for Workloads and Image Registry to verify before merging.

:crossed_fingers: the build tests get unblocked today.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (flavianmissi) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:170]: sounds good to me. do we want image registry QE to take this for a spin?

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (ingvagabund) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:158]: There's also `openshift.io/deployer` [1] that needs to be disabled. 
 [1] https://github.com/openshift/api/blob/88a8afea030c56055c81629b567fdcb5ab6a3903/openshiftcontrolplane/v1/types.go#L208

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (ingvagabund) [reply pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:170]: https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/configobservation/controllers/capability_imageregistry.go includes `openshift.io/serviceaccount-pull-secrets` as well.

File: pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go
Comment (ingvagabund) [inline pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go:145]: https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/configobservation/controllers/capability_builds.go#L24-L27 for the full list. `openshift.io/builder-serviceaccount` is missing.

**Comment by apoorvajagtap** [comment]
Thanks @ingvagabund! I have mapped the respective SA controllers to the [controllerCapabilities](https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/335/files#diff-8373bcc704ec98a02699a189cb85002fb2b70a3f6812c7dc02a0d529c606f6f3) too. PTAL.

**Comment by ingvagabund** [comment]
LGTM

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by adambkaplan** [comment]
/label px-approved

No substantial change to product experience with this update.

@apoorvajagtap please draft a release note for OCPBUILD-9, and then I'll add the "docs-approved" label.

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**
